core/server/services/settings/ensure-settings.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const fs = require('fs-extra'),
+    path = require('path'),
+    debug = require('ghost-ignition').debug('settings:ensure-settings'),
+    _ = require('lodash'),
+    common = require('../../lib/common'),
+    knownSettings = require('./index').knownSettings,
+    config = require('../../config');
+
+/**
+ * Makes sure that all support
+ * @returns {Promise}
+ * @description Returns settings object as defined per YAML files in
+ * `/content/settings` directory.
+ */
+module.exports = function ensureSettingsFiles() {
+    return new Promise(function (resolve, reject) {
+        return _.each(knownSettings, function (setting) {
```

Comment:
I am little bit confused about the logic here 🤔 

You return a Promise and you iterate over the known settings with lodash (sync). The known settings is an array of length n. And for each setting you check if the file exists, which is also a Promise. So you `resolve` the outer error n times, which looks wrong. That also means that the first know settings, will resolve the outer promise. I guess you wanted to use `Promise.each`?

Reply:
*looks up `Promise.each`*.... 👀